src/Lachain.Core/Blockchain/Pool/TransactionPool.cs

```diff
@@ -71,6 +75,50 @@ ITransactionHashTrackerByNonce transactionHashTracker
             _transactionsQueue = new HashSet<TransactionReceipt>();
 
             _blockManager.OnBlockPersisted += OnBlockPersisted;
+            _poolSyncThread = new Thread(PoolSyncWorker);
+            _poolSyncRunning = false;
+            _transactionToSync = new Queue<TransactionReceipt>();
+        }
+
+        private void PoolSyncWorker()
+        {
+            var waitingTime = 500;
+            while (_poolSyncRunning)
+            {
+                lock (_transactionToSync)
+                {
+                    while (_transactionsQueue.Count == 0)
+                        Monitor.Wait(_transactionToSync);
+
+                    var txes = new List<TransactionReceipt>();
+                    while (_transactionToSync.Count > 0)
+                    {
+                        var tx = _transactionToSync.Dequeue();
+                        txes.Add(tx);
+                    }
+                    TransactionAdded?.Invoke(this, txes);
+                }
+                // wait for several txes to appear, so we send several txes together in pool-sync
+                Thread.Sleep(waitingTime);
```

Comment:
We should resend txes as fast as we get it,  no waiting . If we have only one tx we should send this tx only

Reply:
Then how you suggest we send txes in batch? because this way always one tx will be sent at a time.